app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +329,117 @@ angular.module('confRegistrationWebApp')
+    }
+
+    // Create a new registrant on the server
+    // Returns a promise the resolves when the registrant has been created
+    function createRegistrant (registrant) {
+      return $http.put('/registrants/' + registrant.id, registrant);
+    }
+
+    // Take the current registration and merge it into the spouse's registration
+    function mergeWithSpouse () {
+      // Generate an array of new registrants that include all attributes
+      var newRegistrants = $scope.currentRegistration.registrants.map(function (registrant) {
+        var newRegistrantId = uuid();
+
+        // Make a copy of the answers, but overwrite the id and registrantId attributes
+        var answers = registrant.answers.map(function(answer) {
+          return _.assign({}, answer, {
+            id: uuid(),
+            registrantId: newRegistrantId
+          });
+        });
+
+        return {
+          id: newRegistrantId,
+          registrationId: $scope.spouseRegistration.id,
+          registrantTypeId: registrant.registrantTypeId,
+          answers: answers,
+          firstName: registrant.firstName,
+          lastName: registrant.lastName,
+          email: registrant.email
+        };
+      });
+
+      //Payload for new spouse registration
+      var newSpouseRegistration = {
+        id: $scope.spouseRegistration.id,
+        conferenceId: $scope.currentRegistration.conferenceId,
+        registrants: newRegistrants.map(function (registrant) {
+          // When creating a new registration, only a few registrant attributes are required, so only keep a few of the
+          // registrant attributes
+          var sparseRegistrant = _.pick(registrant, ['id', 'registrationId', 'registrantTypeId']);
+          sparseRegistrant.answers = [];
+          return sparseRegistrant;
+        })
+      };
+
+      //Add new registration
+      return createRegistration(newSpouseRegistration)
+        .then(function () {
+          //Add registrants to new registration
+          // Advance to the next step after all the registrations have been created
+          return $q.all(newRegistrants.map(createRegistrant));
+        }).then(function () {
+          //Delete existing registration
+          return deleteRegistration($scope.currentRegistration);
+        }).then(function () {
+          // Reload the merged spouse registration
+          return loadRegistration($scope.spouseRegistration.id);
+        }).then(function (mergedRegistration) {
+          // Update the UI to show the merged registration because it includes all of the registrants
+          $scope.currentRegistration = mergedRegistration;
+
+          // Hide certain elements and sections in the UI because the current user is not able make changes to their
+          // spouses registration, even though they are now a registrant on that registration
+          $scope.mergedRegistration = true;
+        }).catch(errorFromResponse('An error occurred while merging spouse registrations.'));
```

Comment:
What happens if this fails partway through? You could end up with extra registrations or registrants if you don't get past the delete step. Idk how big of an issue that is.